kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/MessageFetchContext.java

```diff
@@ -171,9 +176,11 @@ private void addErrorPartitionResponse(TopicPartition topicPartition, Errors err
         tryComplete();
     }
 
-    private synchronized void tryComplete() {
-        if (responseData.size() >= fetchRequest.fetchData().size()) {
-            complete();
+    private void tryComplete() {
+        if (responseData != null && responseData.size() >= fetchRequest.fetchData().size()) {
```

Comment:
The null check for `responseData` here might not be thread safe without `synchronized`. Should we add `volatile` keyword to `responseData`?

Reply:
@BewareMyPower good, I replace responseData with resultFuture to judge the null pointer exception, because resultFuture with volatile